### ISSUE TITLE
Error with RecordDelete generioc handler during tutorial

### ISSUE THREAD

**Issue by lebogan** [martenframework/marten#36]
After creating a few articles, I can delete them successfully with:
```
class ArticleDeleteHandler < Marten::Handler
  @article : Article?

  def get
    render("article_delete.html", context: {article: article})
  end

  def post
    article.delete
    redirect(reverse("home"))
  end

  private def article
    @article ||= Article.get!(id: params["pk"])
  rescue Marten::DB::Errors::RecordNotFound
    raise Marten::HTTP::Errors::NotFound.new("Article not found")
  end
end
```

When I switch the code to the generic handler:
```
class ArticleDeleteHandler < Marten::Handlers::RecordDelete
  model Article
  template_name "article_delete.html"
  success_route_name "home"
end
```
I get an error:
```
Marten::Template::Errors::UnknownVariable at /article/2/delete
19-24 minutes
Snippet

lib/marten/src/marten/template/variable.cr

42                      context[bit]
43                    else
44                      current.not_nil![bit]
45                    end
46        rescue KeyError | Errors::UnknownVariable
47          raise Errors::UnknownVariable.new(
48            String.build do |s|
49              s << "Failed lookup for attribute '"
50              s << bit.to_s
51              s << "'"
52              s << " in " + current.to_s if current.is_a?(Context)
```            
![MartenTemplateErrorsUnknownVariable at article2delete](https://user-images.githubusercontent.com/1024382/197868381-3fe037b4-164f-4508-844b-01b68890ae3c.png)

Am I missing something in my code?

**Comment by ellmetha** [comment]
Hey! 👋 Thanks for pointing this out!

Indeed, there is a missing call to `#record_context_name` in the `ArticleDeleteHandler`:

```crystal
class ArticleDeleteHandler < Marten::Handlers::RecordDelete
  model Article
  template_name "article_delete.html"
  success_route_name "home"
  record_context_name "article"
end
```

I will fix this and update the documentation!

**Comment by ellmetha** [comment]
Fixed in a89bacc. Thanks again!
